src/hooks/http.ts

```diff
@@ -64,6 +64,9 @@ function handleClientRequest(request: http.ClientRequest) {
 
   const startTime = getTime();
   request.on("finish", () => {
+    // recording may have finished at this point
+    if (!recording.running) return;
```

Comment:
We should probably show a warning here — this might point to a problem with the instrumented test.